airbyte-integrations/connectors/source-shopify/source_shopify/utils.py

```diff
@@ -92,34 +130,83 @@ def get_wait_time(*args, threshold: float = 0.9, rate_limit_header: str = "X-Sho
             load = int(current_rate) / int(max_rate_limit)
         else:
             load = None
-        # define wait_time based on load conditions
-        if not load:
-            # when there is no rate_limits from header, use the `sleep_on_unknown_load`
-            wait_time = ShopifyRateLimiter.on_unknown_load
-        elif load >= threshold:
-            wait_time = ShopifyRateLimiter.on_high_load
-        elif load >= mid_load:
-            wait_time = ShopifyRateLimiter.on_mid_load
-        elif load < mid_load:
-            wait_time = ShopifyRateLimiter.on_low_load
+        wait_time = ShopifyRateLimiter._convert_load_to_time(load, threshold)
```

Comment:
seems like load here is type `float` but an `Optional[int]` is expected in the function

Reply:
Good spot! Updated the expected type to be `Optional[float]`